source-code-git/fs/src/schema.ts

```diff
@@ -25,6 +25,5 @@ export type Filesystem = {
 		 *   fs.toJson({ exclude: ["**\/node_modules"] })
 		 */
 		exclude?: string[]
-	}) => Record<string, any>
-	fromJson: (json: string) => Filesystem
+	}) => Promise<Record<string, any>>
```

Comment:
`fromJson` should return an instance of the filesystem, not a Record. 

Here is a real world example https://github.com/inlang/inlang/blob/main/source-code/rpc/src/functions/generateConfigFile.ts#L60

Reply:
Woops, that was a mistake, probably made when I was shuffling type declarations around :P

Reply:
The diff is confusing, but that is the return value for `toJson`. I removed `fromJson` from the type declaration as I had implemented it as a static method on the class.